src/ExportModule.ts

```diff
@@ -32,10 +32,21 @@ export function exportModule(libName: string, dataTypes: CalculatorTypes.DataTyp
 
   dataTypes.forEach((object, i) => {
     const stateName = `${object.dataType}_${i}`;
+
     if (object.dataType !== null && factory(object.dataType, stateName) !== null) {
-      const StateClass = factory(object.dataType, stateName);
+      const stateClass = factory(object.dataType, stateName);
+      if (object.valueSet !== undefined && stateClass && states.doesHaveCodes(stateClass)) {
```

Comment:
Why not simply try the getter you added?
```suggestion
      if (object.valueSet !== undefined && stateClass && stateClass.hasCodesField) {
```

Reply:
```suggestion
      if (object.valueSet !== undefined && stateClass?.hasCodesField)) {
```
Even better suggestion.